omod/src/main/java/org/openmrs/module/dhisreport/web/controller/Dhis2ServerController.java

```diff
@@ -99,25 +110,56 @@ public void editReportCode( @RequestParam( required = false, value = "reportCode
 
         DHIS2ReportingService service = Context.getService( DHIS2ReportingService.class );
         ReportDefinition rd = service.getReportDefinition( reportId );
-        rd.setCode( reportCode );
-        service.saveReportDefinition( rd );
 
     }
 
-    @RequestMapping( value = "/module/dhisreport/configureDhis2", method = RequestMethod.POST )
+    @RequestMapping( value = "/module/dhisreport/configureDhis2", params = "checkConnection", method = RequestMethod.POST )
+    public String CheckConnectionWithDHIS2( ModelMap model, WebRequest webRequest, HttpServletRequest request )
+    {
+        DHIS2ReportingService service = Context.getService( DHIS2ReportingService.class );
+        HttpDhis2Server server = service.getDhis2Server();
+
+        String dhisurl = Context.getAdministrationService().getGlobalProperty( "dhisreport.dhis2URL" );
+        String dhisusername = Context.getAdministrationService().getGlobalProperty( "dhisreport.dhis2UserName" );
+        String dhispassword = Context.getAdministrationService().getGlobalProperty( "dhisreport.dhis2Password" );
+
+        URL url = null;
+        try
+        {
+            url = new URL( dhisurl );
+        }
+        catch ( MalformedURLException e )
+        {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+        boolean val = testConnection( url, dhisusername, dhispassword, server, webRequest, model );
+
+        String referer = webRequest.getHeader( "Referer" );
+
+        if ( val == false )
+        {
+            request.getSession().setAttribute( "ConnectionErrorMessage", "DHIS2 Connection Not Established" );
+            return "redirect:" + referer;
+        }
+        webRequest.setAttribute( WebConstants.OPENMRS_MSG_ATTR, Context.getMessageSourceService().getMessage(
+            "dhisreport.dhis2ConnectionSuccess" ), WebRequest.SCOPE_SESSION );
+        return "redirect:" + referer;
+    }
+
+    @RequestMapping( value = "/module/dhisreport/configureDhis2", method = RequestMethod.POST, params = "submit" )
     public void saveConfig( ModelMap model, @RequestParam( value = "url", required = true )
```

Comment:
@puevigreven , The task of "test connection" is separate from "configure dhis2", which this controller is performing, you should not perform this task here as having atomic actions will make it more readable and will help it to be replicated for a REST call or any such task without having to "configureDhis2".